addons/event_booth_sale/static/src/js/event_booth_configurator_widget.js

```diff
@@ -0,0 +1,148 @@
+     * @returns {Promise<Boolean>} stopPropagation true if a suitable configurator has been found.
+     *
+     * @override
+     * @private
+     */
+    _onProductChange: function (productId, dataPointId) {
+        return this._super.apply(this, arguments).then((stopPropagation) => {
+            if (stopPropagation) {
+                return Promise.resolve(true);
+            } else {
+                return this._isProductEventBooth(productId, dataPointId);
+            }
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //-------------------------------------------------------------------------
+
+    /**
+     * This method will check if the productId needs a configuration or not.
+     *
+     * @param {integer} productId: product.product ID
+     * @param {String} dataPointId
+     * @returns {Promise<Boolean>} stopPropagation true if the product is an event booth.
+     *
+     * @private
+     */
+    _isProductEventBooth: function (productId, dataPointId) {
+        const self = this;
+        return this._rpc({
+            model: 'product.product',
+            method: 'read',
+            args: [productId, ['is_event_booth']],
+        }).then(function (result) {
+            if (result && result[0].is_event_booth) {
+                self._openEventBoothConfigurator({
+                    default_product_id: productId
+                }, dataPointId);
+                return Promise.resolve(true);
+            }
+            return Promise.resolve(false);
+        });
+    },
+
+    /**
+     * Opens the event booth configurator allowing to configure the SO line with its informations.
+     *
+     * When the window is closed, configured values are used to trigger a 'field_changed'
+     * event to modify the current SO line.
+     *
+     * If the window is closed without providing the required values 'event_id', 'event_booth_category_id'
+     * and 'event_booth_pending_ids', the product_id field is cleaned.
+     *
+     * @param {Object} add_context: various "default_" values added to context when calling
+     *  configuration wizard model;
+     * @param dataPointId
+     *
+     * @private
+     */
+    _openEventBoothConfigurator: function (add_context, dataPointId) {
+        const self = this;
+        this.do_action('event_booth_sale.event_booth_configurator_action', {
+            additional_context: add_context,
+            on_close: function (result) {
+                if (result && !result.special) {
+                    self.trigger_up('field_changed', {
+                        changes: result.eventBoothConfiguration,
+                        dataPointID: dataPointId,
+                        onSuccess: function () {
+                            self._onLineConfigured();
+                        }
+                    });
+                } else if (!self.recordData.event_id || !self.recordData.event_booth_ids) {
```

Comment:
Does recordData refer to a sale.order.line here? (I think so).
Then, since the wizard sets "event_booth_pending_ids", not sure this condition is accurate.
(Or maybe I'm missing something).